src/js/chartAuthoring.js

```diff
@@ -144,4 +147,19 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         });
         return pieChartData;
     };
+
+    // Adds aria attributes that only make sense within the context of the overall chart authoring tool
+    // Specifically, adds:
+    // - a unique ID for the pie's SVG and the legend's table
+    // - an aria-controls attribute for the dataEntryPanel's form referencing the unique IDs 
+    floe.chartAuthoring.addAriaConnections = function(that) {
+        var legendId = "floe-id-"+fluid.allocateGuid(),
```

Comment:
In the case that users customize the template to provide id attributes for legend of pie elements, we probably should use those id values directly instead of overwriting them with generated ids. We probably should add a function that checks whether the element already has id attribute, if yes, returns its value; if not, adds it and returns the generated id value. This function can be used for line 156, 157, as well as line 179, 180 in pie.js.